multisite-directory.php

```diff
@@ -246,6 +264,16 @@ public static function dashboard_glance_items ($items) {
         return $items;
     }
 
+    /**
+     * Redirects single site post to the actual site
+     */
+    public static function template_redirect () {
+        if (is_singular(Multisite_Directory_Entry::name)) {
```

Comment:
You're right, I completely overlooked that! I'm removing this for now and will open the issue, but your solution sounds just perfect!